Packages/com.unity.inputsystem/InputSystem/Plugins/Switch/SwitchProControllerHID.cs

```diff
@@ -113,12 +129,352 @@ namespace UnityEngine.InputSystem.Switch
     /// <summary>
     /// A Nintendo Switch Pro controller connected to a desktop mac/windows PC using the HID interface.
     /// </summary>
-    /// <remarks>
-    /// This controller currently only works when connected via Bluetooth but not when connected over USB.
-    /// </remarks>
     [InputControlLayout(stateType = typeof(SwitchProControllerHIDInputState), displayName = "Switch Pro Controller")]
-    public class SwitchProControllerHID : Gamepad
+    public class SwitchProControllerHID : Gamepad, IInputStateCallbackReceiver, IEventPreProcessor
     {
+        protected override void OnAdded()
+        {
+            base.OnAdded();
+            HandshakeRestart();
+        }
+
+        private static readonly SwitchMagicOutputReport.CommandIdType[] s_HandshakeSequence = new[]
+        {
+            SwitchMagicOutputReport.CommandIdType.Status,
+            SwitchMagicOutputReport.CommandIdType.Handshake,
+            SwitchMagicOutputReport.CommandIdType.Highspeed,
+            SwitchMagicOutputReport.CommandIdType.Handshake,
+            SwitchMagicOutputReport.CommandIdType.ForceUSB
+            ////TODO: Should we add a step to revert back to simple interface?
```

Comment:
should we move this todo to a jira ticket to track?

Reply:
ISX-957